src/cli-constant.js

```diff
@@ -158,6 +158,13 @@ const detailedOptions = normalizeDetailedOptions({
   "debug-print-doc": {
     type: "boolean"
   },
+  editorconfig: {
+    type: "boolean",
+    category: CATEGORY_CONFIG,
+    description: "Take .editorconfig into account when parsing configuration.",
+    forwardToApi: true,
```

Comment:
IMO, it shouldn't be forwarded to API just like `--config`, it's a CLI-only flag.

Reply:
I'm sure if you're saying that it *is* a CLI-only flag, or that it *shouldn't be*, but it isn't currently. Even though the default behavior only affects the CLI usage, I think we should leave the API option in place, so that editor integrations can opt-in to supporting `.editorconfig` without having to shell out to the CLI.

Reply:
I mean, it should be both a CLI flag (`--no-editorconfig` for backward compatibility I guess?) and an option in API's `resolveConfig`, but not an option in `.prettierrc`.

Sorry for the confusing.

Reply:
Ohh, you're saying that it will be parsed from `.prettierrc`? I just did a quick test, and it seemed not to be...

```bash
set -x
node ./bin/prettier.js tests_integration/cli/config/editorconfig/file.js
echo 'editorconfig: false' >> tests_integration/cli/config/.prettierrc
node ./bin/prettier.js tests_integration/cli/config/editorconfig/file.js
node ./bin/prettier.js --no-editorconfig tests_integration/cli/config/editorconfig/file.js
git checkout tests_integration/cli/config/.prettierrc
set +x
```

Reply:
Oh, I though you wrote that way. Then you don't need `forwardToApi: true` which is for forwarding to prettier.format.options (`.prettierrc`), and so does `options.js`.

https://github.com/josephfrazier/prettier/blob/64a441384dafc6dec3d51e978dc068b6bbfb585f/src/options.js#L26

Reply:
Ah, thanks for the explanation! I'm not too familiar with how the current options system is set up, so I just copied code from the other options. Fixed in https://github.com/prettier/prettier/pull/3255/commits/53013cbdc0fe633d6094cb97bfdd165afe39b137